cmake/extensions.cmake

```diff
@@ -2092,23 +2094,42 @@ endfunction()
 #                                  BUILD debug, will look for <board>_debug.conf
 #                                  and <board>_debug.overlay, instead of <board>.conf
 #
-# returns an updated list of absolute paths
+# FILENAMES <list>: Construct a filename based on the provided BOARD, REVISION, and BUILD type.
+#                   FILENAMES takes the following additional arguments:
+#                   BOARD <board>:             Board name to use when constructing FILENAMES list.
+#                   BOARD_REVISION <revision>: Board revision to use when constructing FILENAMES list.
+#                   BUILD <type>:               Build type to use when constructing FILENAMES list.
```

Comment:
These three sentences provide no additional information.

Either re-write them or remove them.

Reply:
Sorry, I meant just the sentences explaining what the arguments were.

Reply:
@SebastianBoe do you think it would be better to describe those arguments which are dependent on the first argument independently ?

So that `BOARD`, `BOARD_REVISION`, and `BUILD` are only described once.

Currently I was more following similar style as here: https://cmake.org/cmake/help/latest/command/file.html
where each sub-command describes it's own arguments even if it means those arguments has already been described for an earlier sub-command.

Reply:
IMHO we shouldn't use a modal function for FILENAMES at all.

Reply:
> IMHO we shouldn't use a modal function for FILENAMES at all.

because ?
A good reasoning will be appreciated, it's a PR so not to late to change.

Reply:
Modal functions increase the length of functions and the amount of mutable state. Making them harder to read, review and verify.

Reply:
> Modal functions increase the length of functions and the amount of mutable state. Making them harder to read, review and verify.

On a generally note I agree with you (which is why I hoped for a more specific reason).
But with this particular case, I feel the `zephyr_file()` should stay close to the CMake `file()` which uses this principle.
Being aligned with CMake's `file()` is why I proposed this particular scheme here. (To me consistency wins)

Reply:
I disagree, but I will let you do as you wish as the maintainer :)

Reply:
> I disagree, but I will let you do as you wish as the maintainer

Just marked `dnm` to allow some more thoughts.
I'm not really happy with the modal approach but I also like to be somewhat aligned with the approach laid out by CMake `file()`

Reply:
excessive space
s/ Build type/Build type